src/main/java/com/github/susom/database/StatementAdaptor.java

```diff
@@ -80,6 +82,8 @@ public void addParameters(PreparedStatement ps, Object[] parameters) throws SQLE
         } else {
           ps.setNull(i + 1, sqlNull.getType());
         }
+      } else if (parameter instanceof java.sql.Date) {
+        ps.setDate( i + 1, (java.sql.Date) parameter);
```

Comment:
See longer comments in CommonTest. Probably requires lots of testing, but at least in some cases like Oracle, I would be tempted to explicitly set the time component to noon to make accidental date shifts less likely.

Reply:
Yes - my use case is Postgres, not Oracle, but I recognize we ned to support that.  I knew Oracle had a DateTime type, and did not realize they were stashing timestamps on Date. Thank you for pointing that gotcha out. 

For the sleep survey prototype, the reason I was doing this work, we have generally been using the server time zone for our data.  Do you consider this acceptable for things like DoB, with perhaps an option to override this? 

I will see about modifying the code to force the time to noon for Oracle on insert/update for our dates, and add some testing for it.

I'm not sure how far you want me to take this implementation?  I undertook this mostly so I could get the sleep survey up and running with susom backend standards like vert.x and database.   The noon timestamp should meet our requirements, but that may not be acceptable.  Do we need to talk further design?

Reply:
BTW, I cannot set the time to noon with LocalDate - there is no time component.  And LocalDateTime has no notion of Timezone either.  I would have to convert these things to Timestamps to do this.   

I am really sure how helpful this would be.    
Oracle and SQLServer 0 out the time related fields including TimeZone when given a LocalDate.  Postgres actually has a decent Date implementation in their driver.  

The only DB that does not is HSQL.  It 0's out time, but puts a timezone on everything.  As a result, if you change timezone, it fails to match LocalDate/LocalDateTime values.  It's a documented crappy implementation and I don't see a clean way to work around it.  Even if I set time to noon, it will fail to match if the client is in a different timezone.  At one point, I tried setting it to noon with UTC, but it really doesn't help.  Noon UTC means it would change a date like 8/29 to 8/28 on our systems, and then the roundtrip lookup still fails because adding a timezone of PST to 8/28 just gives us an 8/28 date.

Reply:
BTW, I added a test that tries every timezone in common test.  Current implementation is returning the date I stored, for Derby, Postgres, Oracle, and SQL Server.  Only HSQL fails this.